src/openrct2/localisation/Localisation.h

```diff
@@ -98,13 +98,41 @@ template<typename T> void constexpr set_format_arg(uint8_t* args, size_t offset,
         set_format_arg_body(gCommonFormatArgs, offset, (uintptr_t)(value), sizeof(type));                                      \
     } while (false)
 
-#define set_format_arg_on(args, offset, type, value) set_format_arg_body(args, offset, (uintptr_t)(value), sizeof(type))
-
 #define set_map_tooltip_format_arg(offset, type, value)                                                                        \
     do                                                                                                                         \
     {                                                                                                                          \
         static_assert(sizeof(type) <= sizeof(uintptr_t), "Type too large");                                                    \
         set_format_arg_body(gMapTooltipFormatArgs, offset, (uintptr_t)(value), sizeof(type));                                  \
     } while (false)
 
+class Formatter
+{
+    const uint8_t* StartBuf;
+    uint8_t* CurrentBuf;
+
+public:
+    explicit Formatter(uint8_t* buf)
+        : StartBuf(buf)
+        , CurrentBuf(buf)
+    {
+    }
+
+    auto buf()
+    {
+        return CurrentBuf;
+    }
+
+    std::size_t bytes() const
+    {
+        return CurrentBuf - StartBuf;
+    }
+
+    template<typename TSpecified, typename TDeduced> Formatter& add(TDeduced value)
+    {
+        set_format_arg_body(CurrentBuf, 0, (uintptr_t)(value), sizeof(TSpecified));
```

Comment:
static cast

Reply:
I can't static cast because sometimes it needs reinterpret cast:
```
D:\github\OpenRCT2\src\openrct2\localisation\Localisation.h(132,1): error C2440: 'static_cast': não é possível converter de 'TDeduced' para 'uintptr_t' [D:\github\OpenRCT2\src\openrct2\libopenrct2.vcxproj]
          with
          [
              TDeduced=char *
          ] (compilando arquivo de origem network\Network.cpp)
```
I can't reinterpret_cast because compiler says that it's ok to use static cast:

```
D:\github\OpenRCT2\src\openrct2\localisation\Localisation.h(132,1): error C2440: 'reinterpret_cast': não é possível converter de 'TDeduced' para 'uintptr_t' [D:\github\OpenRCT2\src\openrct2\libopenrct2.vcxproj]
          with
          [
              TDeduced=money32
          ] (compilando arquivo de origem actions\GameAction.cpp)
...
D:\github\OpenRCT2\src\openrct2\localisation\Localisation.h(132,44): message : Conversão é uma conversão padrão válida, que pode ser realizada implicitamente ou pelo uso de static_cast, conversão C-style ou con
versão function-style (compilando arquivo de origem actions\GameAction.cpp) [D:\github\OpenRCT2\src\openrct2\libopenrct2.vcxproj]
```

Reply:
Oh, that's in portuguese.
The first one says it can't converte `char*` to `uintptr_t` without reinterpret cast.
The second one says it won't convert `money32` ot `uintptr_t` using reinterpret cast because static_cast will do the job.

Make up your mind compiler!

Reply:
It looks to me like there are two options out of this:
1. Template-fu
2. Comment
I will leave the decision up to you.

Reply:
I added a comment. I'm not entirely sure what you meant with 'template-fu'. Anyway, there is still more to be done in this code, like adding some static asserts and removing more macros if possible. I'll surely get back to this.

Reply:
Found a way to solve it :)